src/Twig/Runtime/UrlExtensionRuntime.php

```diff
@@ -252,13 +252,17 @@ public function postCommentDeleteUrl(PostComment $comment): string
         ]);
     }
 
-    public function optionsUrl(string $name, string $value, string $routeName = null): string
+    // Hopefully $additionalParams is temporary
+    public function optionsUrl(string $name, string $value, string $routeName = null, array $additionalParams = []): string
```

Comment:
I think currently $additionalParams is only used in one very specific case (header nav bar for per-magazine view), and the hope is that that use case will go away with future UI changes (unification of microblog and theads content) and so this change won't need to stay around.

There's nothing particularly wrong with needing to use optionsUrl() to add multiple new options, just it'd be cleaner to avoid adding this rarely-used codepath and I think in the near future it'll go back away again.

Reply:
So, I made it worse (with the current iteration we're wanting to use optionsUrl() to clear out the ?p= parameter when we change the filters, so we go back to the first page of results for the new filter settings, so it may not be so temporary anymore). See my comment below.